Java/core/src/main/java/com/amazon/randomcutforest/util/ArrayPacking.java

```diff
@@ -78,37 +113,63 @@ static int logMax(long base) {
+     *                    with arithmetic compression enabled.
+     * @return the array of unpacked ints.
+     */
+    public static int[] unpackInts(int[] packedArray, boolean decompress) {
+        checkNotNull(packedArray, " array unpacking invoked on null arrays");
+
+        if (!decompress) {
+            return Arrays.copyOf(packedArray, packedArray.length);
+        }
 
-        return (inputArray.length < 3) ? unPackInts(inputArray, inputArray.length, decompress)
-                : unPackInts(inputArray, inputArray[2], decompress);
+        return (packedArray.length < 3) ? unpackInts(packedArray, packedArray.length, decompress)
+                : unpackInts(packedArray, packedArray[2], decompress);
     }
 
-    public static int[] unPackInts(int[] inputArray, int length, boolean decompress) {
-        checkNotNull(inputArray, " array unpacking invoked on null arrays");
+    /**
+     * Unpack an array previously created by {@link #pack(int[], int, boolean)}.
+     * 
+     * @param packedArray An array previously created by
+     *                    {@link #pack(int[], int, boolean)}.
+     * @param length      The desired length of the output array. If this number is
+     *                    different from the length of the array that was originally
+     *                    packed, then the result will be truncated or padded with
+     *                    zeros as needed.
+     * @param decompress  A flag indicating whether the packed array was created
+     *                    with arithmetic compression enabled.
+     * @return the array of unpacked ints.
+     */
+    public static int[] unpackInts(int[] packedArray, int length, boolean decompress) {
+        checkNotNull(packedArray, " array unpacking invoked on null arrays");
         checkArgument(length >= 0, "incorrect length parameter");
 
-        if (inputArray.length < 3 || !decompress) {
-            return Arrays.copyOf(inputArray, length);
+        if (packedArray.length < 3 || !decompress) {
+            return Arrays.copyOf(packedArray, length);
         }
-        int min = inputArray[0];
-        int max = inputArray[1];
+        int min = packedArray[0];
+        int max = packedArray[1];
         int[] output = new int[length];
         if (min == max) {
-            if (inputArray[2] >= length) {
+            if (packedArray[2] >= length) {
                 Arrays.fill(output, min);
             } else {
-                for (int i = 0; i < inputArray[2]; i++) {
+                for (int i = 0; i < packedArray[2]; i++) {
                     output[i] = min;
                 }
             }
         } else {
             long base = ((long) max - min + 1);
             int packNum = logMax(base);
             int count = 0;
-            for (int i = 3; i < inputArray.length; i++) {
-                long code = inputArray[i];
+            for (int i = 3; i < packedArray.length; i++) {
+                long code = packedArray[i];
```

Comment:
The next line should be min(packedArray[2],output.length).  But we can fix in the next PR as well.

Reply:
Fixed.